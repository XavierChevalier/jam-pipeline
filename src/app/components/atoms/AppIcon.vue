<script lang="ts">
  import { defineComponent } from 'vue'
  import { SizeMixin } from '@/app/mixins/SizeMixin'

  export default defineComponent({
    mixins: [
      SizeMixin({
        sm: 'w-4.5 h-4.5',
        base: 'w-6 h-6',
        lg: 'w-7 h-7',
        xl: 'w-9 h-9',
      }),
    ],
  })
</script>

<script setup lang="ts">
  import { computed, PropType } from 'vue'
  import { MdiExtra, MdiExtraObject } from '@/assets/images/icons/MdiExtra'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'

  const props = defineProps({
    path: {
      type: [String, Object] as PropType<MdiExtra>,
      required: true,
      validator: isPropertyValid(MdiExtra),
    },
  })
  const pathObject = computed<MdiExtraObject>(() =>
    typeof props.path === 'object' ? props.path : { path: props.path }
  )
</script>

<template>
  <svg
    class="inline-block align-middle"
    :class="sizeClass"
    :viewBox="pathObject.viewBox || '0 0 24 24'"
  >
    <path
      :transform="pathObject.transform"
      :d="pathObject.path"
      style="fill: currentColor"
    />
  </svg>
</template>
