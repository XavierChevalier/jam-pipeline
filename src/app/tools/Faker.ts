export const generateImageUrl = (size = 50) => `https://picsum.photos/${size}`
