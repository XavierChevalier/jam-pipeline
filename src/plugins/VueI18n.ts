import { createI18n } from 'vue-i18n'
import { Ref } from 'vue'

export const i18n = createI18n({
  legacy: false,
  globalInjection: true,
  locale: 'en',
  fallbackLocale: 'en',
})

export function changeLocale(locale: 'en' | 'fr') {
  if (i18n.mode === 'legacy') {
    i18n.global.locale = locale
    return
  }

  ;(i18n.global.locale as unknown as Ref<string>).value = locale
}
