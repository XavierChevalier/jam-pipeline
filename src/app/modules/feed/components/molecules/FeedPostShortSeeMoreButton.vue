<script setup lang="ts">
  import { mdiNewsOutline } from '@/assets/images/icons/MdiExtra'
  import { mdiChevronDown } from '@mdi/js'
  import { useI18n } from 'vue-i18n'
  import AppIcon from '@/app/components/atoms/AppIcon.vue'

  const { t } = useI18n()
</script>

<template>
  <div class="text-neutral text-center">
    <AppIcon size="sm" :path="mdiNewsOutline" />
    <span class="pl-2 align-middle">{{ t('seeMore') }}</span>
    <div>
      <AppIcon size="sm" :path="mdiChevronDown" />
    </div>
  </div>
</template>

<i18n lang="yaml">
en:
  seeMore: 'See all news'
fr:
  seeMore: 'Voir toutes les news'
</i18n>
