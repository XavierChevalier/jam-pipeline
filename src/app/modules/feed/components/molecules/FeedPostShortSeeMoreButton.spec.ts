import FeedPostShortSeeMoreButton from './FeedPostShortSeeMoreButton.vue'
import * as Stories from '@/app/modules/feed/components/molecules/FeedPostShortSeeMoreButton.stories'
import { changeLocale } from '@/plugins/VueI18n'
import AppIcon from '@/app/components/atoms/AppIcon.vue'
import { shallowMountWithPlugins } from '@/tests/tools/MountWithPlugins'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'

describe('FeedPostShortSeeMoreButton', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories).snapshotEachStories()

  it('should contains 2 icons', () => {
    const wrapper = shallowMountWithPlugins(FeedPostShortSeeMoreButton)

    expect(wrapper.findAllComponents(AppIcon)).toHaveLength(2)
  })

  it.each`
    lang    | expectedDisplayText
    ${'fr'} | ${'Voir toutes les news'}
    ${'en'} | ${'See all news'}
  `(
    'should display a text "$expectedDisplayText" in "$lang"',
    ({ lang, expectedDisplayText }) => {
      changeLocale(lang)
      const wrapper = shallowMountWithPlugins(FeedPostShortSeeMoreButton)

      expect(wrapper.text()).toContain(expectedDisplayText)
    }
  )
})
