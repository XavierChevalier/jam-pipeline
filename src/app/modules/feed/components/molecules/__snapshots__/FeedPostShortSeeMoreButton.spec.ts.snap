// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedPostShortSeeMoreButton stories snapshots [snapshot] should render the "Default" component story correctly 1`] = `
<div
  class="text-neutral text-center"
>
  <anonymous-stub
    path="M 2 3 L 2 18 C 2 19.64497 3.3550302 21 5 21 L 19 21 C 20.64497 21 22 19.64497 22 18 L 22 7 L 20 7 L 20 18 C 20 18.56503 19.56503 19 19 19 C 18.43497 19 18 18.56503 18 18 L 18 3 L 2 3 z M 4 5 L 16 5 L 16 18 C 16 18.388348 16.278986 18.657986 16.416016 19 L 5 19 C 4.4349698 19 4 18.56503 4 18 L 4 5 z M 6 7 L 6 10 L 14 10 L 14 7 L 6 7 z M 6 12 L 6 14 L 14 14 L 14 12 L 6 12 z M 6 16 L 6 18 L 14 18 L 14 16 L 6 16 z"
    size="sm"
  />
  <span
    class="pl-2 align-middle"
  >
    See all news
  </span>
  <div>
    <anonymous-stub
      path="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
      size="sm"
    />
  </div>
</div>
`;
